packages/adapter-edgedb/src/index.ts

```diff
@@ -0,0 +1,258 @@
+          id: true,
+          email: true,
+          emailVerified: true,
+          image: true,
+          name: true,
+          filter_single: {
+            id: updatedUser.id,
+          },
+        }))
+        .run(client);
+
+      if (user) {
+        return user;
+      } else {
+        throw new Error("could not find updated user");
+      }
+    },
+    async deleteUser(id) {
+      const query = e.delete(e.User, () => ({
+        filter_single: { id },
+      }));
+
+      await query.run(client);
+    },
+    async linkAccount({
+      userId,
+      type,
+      provider,
+      providerAccountId,
+      refresh_token,
+      access_token,
+      expires_at,
+      token_type,
+      scope,
+      id_token,
+      session_state,
+    }) {
+      const query = e.insert(e.Account, {
+        userId,
+        type,
+        provider,
+        providerAccountId,
+        refresh_token,
+        access_token,
+        expires_at,
+        token_type,
+        scope,
+        id_token,
+        session_state,
+        user: e.select(e.User, () => ({
+          filter_single: { id: userId },
+        })),
+      });
+      await query.run(client);
+
+      return;
+    },
+    async unlinkAccount({ providerAccountId }) {
+      const query = e.delete(e.Account, (account) => ({
+        filter: e.op(account.providerAccountId, "=", providerAccountId),
+      }));
+
+      await query.run(client);
+    },
+    async createSession({ expires, sessionToken, userId }) {
+      const query = e.insert(e.Session, {
+        expires,
+        sessionToken,
+        userId,
+        user: e.select(e.User, () => ({
+          filter_single: { id: userId },
+        })),
+      });
+
+      const result = await query.run(client);
+
+      if (!result) {
+        throw new Error("could not create session");
+      }
+
+      return {
+        expires,
+        sessionToken,
+        userId,
+      };
+    },
+    async getSessionAndUser(sessionToken) {
+      const query = e.select(e.Session, () => ({
+        userId: true,
+        id: true,
+        expires: true,
+        sessionToken: true,
+        user: {
+          id: true,
+          email: true,
+          emailVerified: true,
+          image: true,
+          name: true,
+        },
+        filter_single: { sessionToken },
+      }));
+
+      const sessionAndUser = await query.run(client);
+
+      if (!sessionAndUser) {
+        return null;
+      }
+
+      const { user, ...session } = sessionAndUser;
+
+      if (!user || !session) {
+        return null;
+      }
+
+      return {
+        user,
+        session,
+      };
+    },
+    async updateSession({ sessionToken, expires, userId }) {
+      const query = e.select(
+        e.update(e.Session, () => ({
+          filter_single: { sessionToken },
+          set: {
+            ...(sessionToken && { sessionToken }),
+            ...(expires && { expires }),
+            ...(userId && { userId }),
```

Comment:
I think this should update the `user` link (and the `userId` should be computed)